docs/generalized_distribution.md

```diff
@@ -0,0 +1,83 @@
+keep track of what is distributed and to which partitions it is distributed.
+The actual distribution to the other partitions is done using the `InterPartitionCommandSender`.
+This sends an SBE message to the other partitions containing the command that's being distributed.
+
+Every other partition will write the command to their log and process it. This will be the same
+processor that's used on the distributing partition. As a result we must make sure we don't
+distribute te command again! For this we have the `isCommandDistributed` method on
+the `TypedRecord`. We can detect if a command is distributed by checking the key. Encoded within the
+key is the partition id that generated it. If this doesn't match the partition id that is processing
+the command, it has been distributed before.
+
+At the end of processing the processor must acknowledge the distribution by calling
+the `acknowledgeCommand` method on the `CommandDistributionBehavior`. This method will decode the
+original partition id from the key and, similar to the distribution, will use
+the `InterPartitionCommandSender` to send acknowledgement back to the original partition.
+
+Once the original partition receives this message it writes the `CommandDistribution.ACKNOWLEDGE`
+command to the log. When processing this it knows the distribution to a specific partition is
+completed. Once all partitions acknowledged the distribution the `CommandDistribution.FINISHED`
+event is written. This indicates the finalization of the distribution. At this point all the record
+metadata and raw record value are removed from the state.
+
+### The `CommandDistributionRecord`
+
+The diagram shows the flow of commands and events. What it doesn't highlight is the record that
+belongs to these.
+
+The `CommandDistributionRecord` is relatively simple. It contains 4 properties which are required to
+distribute a command successfully.
+
+|   Property    |                                                                                           Description                                                                                            |
+|---------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Partition id  | The value of the partition id is the id of the partition that has created the record.                                                                                                            |
+| Value type    | The value is part of the record so we know what type of record we are distributing. We need this in combination with the command value to recreate the distributed record.                       |
+| Intent        | The intent is required during redistribution of the command. We must know what intent we should distribute the command with. During normal distribution we can get this from the command itself. |
```

Comment:
🔧 For future readers, it may not be clear what **normal** distribution means. We could choose: 
- to remove this because it's not so much relevant, 
- or to instead clarify that it must be stored in the record -> because it must be stored in the state -> because the redistributor must also decide what intent to send to the other partition